crates/rattler_conda_types/src/conda_lock/mod.rs

```diff
@@ -0,0 +1,201 @@
+use crate::conda_lock::PackageHashes::{Md5, Md5Sha256, Sha256};
+use crate::Platform;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer, Serialize};
+use std::collections::HashMap;
+use url::Url;
+
+const fn default_version() -> u32 {
+    1
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CondaLock {
+    /// Metadata for the lock file
+    pub metadata: LockMeta,
+    /// Locked packages
+    pub package: Vec<LockedDependency>,
+    #[serde(default = "default_version")]
+    pub version: u32,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct LockMeta {
+    /// Hash of dependencies for each target platform
+    pub content_hash: HashMap<Platform, String>,
+    /// Channels used to resolve dependencies
+    pub channels: Vec<Channel>,
+    /// The platforms this lock file supports
+    pub platforms: Vec<Platform>,
+    /// Paths to source files, relative to the parent directory of the lockfile
+    pub sources: Vec<String>,
+    /// Metadata dealing with the time lockfile was created
+    pub time_metadata: Option<TimeMeta>,
+    /// Metadata dealing with the git repo the lockfile was created in and the user that created it
+    pub git_metadata: Option<GitMeta>,
+    /// Metadata dealing with the input files used to create the lockfile
+    pub inputs_metadata: Option<HashMap<String, PackageHashes>>,
+    /// Custom metadata provided by the user to be added to the lockfile
+    pub custom_metadata: Option<HashMap<String, String>>,
+}
+
+/// Stores information about when the lockfile was generated.
+#[derive(Serialize, Deserialize)]
+pub struct TimeMeta {
+    /// Time stamp of lock-file creation format
+    // TODO: I think this is UTC time, change this later, conda-lock is not really using this now
+    pub created_at: String,
+}
+
+/// Stores information about the git repo the lockfile is being generated in (if applicable) and
+/// the git user generating the file.
+#[derive(Serialize, Deserialize)]
+pub struct GitMeta {
+    /// Git user.name field of global config
+    pub git_user_name: String,
+    /// Git user.email field of global config
+    pub git_user_email: String,
+    /// sha256 hash of the most recent git commit that modified one of the input files for this lockfile
+    pub git_sha: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum Manager {
+    #[serde(rename = "conda")]
+    Conda,
+    #[serde(rename = "pip")]
+    Pip,
+}
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct PackageName(String);
+#[derive(Serialize, Deserialize, Eq, PartialEq, Hash)]
+/// This is basically a MatchSpec but will never contain the package name
+pub struct VersionConstraint(String);
+
+#[derive(Serialize)]
+pub enum PackageHashes {
```

Comment:
I am curious why we need this vs. two `Option<String>`?

Also I am wondering if we should deserialize them straight away as the binary representations as properly "typed" checksums or keep them as strings?

Reply:
I just like it that you cannot have the case where both are None :)

Reply:
Yeah so I had the feeling that if both values are none the whole thing should just be excluded from the yaml.

Reply:
Maybe you can add a `serde_with` module to `rattler_digest` to serialize a digests? That would be neet!